### PR TITLE
document lockPermissions and parentID for editChannelPosition

### PR DESCRIPTION
**Pull request by Khaaz**
discord/discord-api-docs#1776

### CODE REVIEW

**Comment by bsian03** [comment]
TS docs needed

File: index.d.ts [hunk 1]
```diff
@@ -1412,7 +1412,7 @@ declare namespace Eris {
       type: string,
       reason?: string
     ): Promise<void>;
-    editChannelPosition(channelID: string, position: number): Promise<void>;
+    editChannelPosition(channelID: string, position: number, options: { lockPermissions: string; parentID: string }): Promise<void>;
```
Comment (bsian03) [inline index.d.ts]: Would be better if we had an interface for the options param

Comment (bsian03) [reply index.d.ts]: Also they should be optional (as well as the other method below)

File: lib/Client.js [hunk 2]
```diff
@@ -918,9 +918,12 @@ class Client extends EventEmitter {
     * Edit a guild channel's position. Note that channel position numbers are lowest on top and highest at the bottom.
     * @arg {String} channelID The ID of the channel
     * @arg {Number} position The new position of the channel
+    * @arg {Object} options Additional options when editing position
+    * @arg {Boolean} options.lockPermissions Whether to sync the permissions with the new parent if moving to a new category
+    * @arg {String} options.parentID The new parent ID (category channel) for the channel that is moved
     * @returns {Promise}
     */
-    editChannelPosition(channelID, position) {
+    editChannelPosition(channelID, position, options) {
```
Comment (bsian03) [inline lib/Client.js]: Could probably deprecate the `position` param and move it into `options` like how we do for `createChannel`

File: lib/Client.js [hunk 3]
```diff
@@ -918,9 +918,12 @@ class Client extends EventEmitter {
     * Edit a guild channel's position. Note that channel position numbers are lowest on top and highest at the bottom.
     * @arg {String} channelID The ID of the channel
     * @arg {Number} position The new position of the channel
+    * @arg {Object} options Additional options when editing position
+    * @arg {Boolean} options.lockPermissions Whether to sync the permissions with the new parent if moving to a new category
+    * @arg {String} options.parentID The new parent ID (category channel) for the channel that is moved
```
Comment (bsian03) [inline lib/Client.js]: All should be optional

File: lib/Client.js (hunk 2 above)
Comment (bsian03) [reply lib/Client.js]: Default param also needed otherwise this'll throw a typeerror

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/Client.js (hunk 2 above)
Comment (Khaaz) [reply lib/Client.js]: position being mandatory anyway, I think we are good keeping there. Avoid an unecessary breaking change as well.
Missed that default param though ty

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**